pkg/router/template/router.go

```diff
@@ -832,7 +832,13 @@ func (r *templateRouter) DeleteEndpoints(id ServiceUnitKey) {
 
 	r.serviceUnits[id] = service
 
-	r.stateChanged = true
+	for alias := range service.ServiceAliasAssociations {
+		if _, exists := r.state[alias]; exists {
+			r.stateChanged = true
+			break
+		}
+	}
```

Comment:
Would
```suggestion
	r.stateChanged = len(frontend.ServiceAliasAssociations) != 0
```
suffice?  That is, is it sufficient to check if there are any associations, without checking that the associations are with things that exist?